bin/xbps-reconfigure/main.c

```diff
@@ -176,10 +187,43 @@ main(int argc, char **argv)
 		rv = xbps_configure_packages(&xh, ignpkgs);
 	} else {
 		for (i = optind; i < argc; i++) {
-			rv = xbps_configure_pkg(&xh, argv[i], true, false);
+			char* pkg = argv[i];
+			if (rdeps) {
+				rv = find_pkg_deps(&xh, pkg, fulldeptree, &deps);
+				if (rv != 0) {
+					fprintf(stderr, "Failed to collect dependencies for "
+						"`%s': %s\n", pkg, strerror(rv));
+				}
+				for (unsigned int j = 0; j < xbps_array_count(deps); j++) {
+					const char *pkgdep = NULL;
+					char pkgname[XBPS_NAME_SIZE];
+					xbps_array_get_cstring_nocopy(deps, j, &pkgdep);
+
+					if (fulldeptree) {
+						if (!xbps_pkg_name(pkgname, sizeof(pkgname), pkgdep)) {
+							fprintf(stderr,
+								"Unable to get package name for dependency `%s'\n", pkgdep);
+							exit(EXIT_FAILURE);
+						}
+					} else {
+						if (!xbps_pkgpattern_name(pkgname, sizeof(pkgname), pkgdep)) {
+							fprintf(stderr,
+								"Unable to get package name for dependency `%s'\n", pkgdep);
+							exit(EXIT_FAILURE);
+						}
+					}
+
+					rv = xbps_configure_pkg(&xh, pkgname, true, false);
+					if (rv != 0) {
+						fprintf(stderr, "Failed to reconfigure "
+							"`%s': %s\n", pkgname, strerror(rv));
+					}
+				}
+			}
+			rv = xbps_configure_pkg(&xh, pkg, true, false);
```

Comment:
Shouldn't the head of the dep tree be the first to be reconfigured?

Reply:
hm, my thinking was dependencies come before the thing that it depends on.

Reply:
Yeah, my brain got stuck on the idea of reverse dependencies hehe
This way looks correct to me!